src/loader/loader_lief_elf.cpp

```diff
@@ -589,15 +590,56 @@ void LoaderLIEF::add_elf_dependencies_to_emulated_fs(
     for (const std::string& lib_name : all_dependencies)
     {
         std::string lib_path = find_library_file(lib_name, libdirs);
-        std::string fs_libdir = "/usr/lib/";
+
         if (lib_path.empty())
         {
             engine->log.warning("Couldn't find library '", lib_name, "': not adding it to emulated filesystem");
             continue;
         }
 
+        // Override FS_LIBDIR if present in virtual_fs
+        std::string fs_libdir = "/usr/lib/";
+
+        if (virtual_fs.find(lib_name) != virtual_fs.end()) {
+            if (virtual_fs.at(lib_name).size() 
+                >= engine->env->fs.get_path_separator().size()) {
+                // If the provided path ends with '/' assume it is a directory
+                if (virtual_fs.at(lib_name).substr(
+                    virtual_fs.at(lib_name).size() 
+                        - engine->env->fs.get_path_separator().size(),
+                    engine->env->fs.get_path_separator().size()
+                ) == engine->env->fs.get_path_separator()) {
+                    // We can just use this as fs_libdir in this case
+                    fs_libdir = virtual_fs.at(lib_name);
+                } else {
+                    // It doesn't end with a '/' so we need to get up to the last occurrence of
+                    // the path separator and use that as the fs_libdir
+                    size_t found = virtual_fs.at(lib_name).find_last_of(
+                        engine->env->fs.get_path_separator());
+
+                    if (found != std::string::npos) {
+                        // We found a path separator, so we can use the string up to that as the
+                        // fs_libdir
+                        fs_libdir = virtual_fs.at(lib_name).substr(0, found);
+                    } else {
+                        // No path separator found, so we should error out here, it needs
+                        // to be an absolute path
+                        throw loader_exception(
+                            "LoaderLIEF::add_elf_dependencies_to_emulated_fs(): "
+                            + virtual_fs.at(lib_name) + " is not an absolute path");
+                    }
+                }
```

Comment:
This code is very similar to the one which loads the binary in `LoaderLIEF::load()`. I think the two could use some factoring, such as a function like below (in pseudo-code):  
```
string get_path_in_virtual_fs(string filename, map<string, string> virtual_fs, string default_dir)
{
    if filename not in virtual_fs:
        return default_dir+path_separator+filename
    else
        path = virtual_fs[filename]
        if path.endswith(path_separator)
               return path+path_separator+filename
        else 
               return path
}
```